src/dams_mcda/WSM.R

```diff
@@ -40,190 +45,386 @@ available_dams <- seq(1:8)
 # list of alternatives
 available_alternatives <- seq(1:5)
 
-matrix_cols <- length(criteria_inputs) # 14 default (output size, adds summedscore)
+matrix_cols <- length(criteria_inputs) # 14 default (output size)
 matrix_rows <- length(available_dams) # 8 default
 matrix_levs <- length(available_alternatives)
 
 
-WSM <- function(RawCriteriaMatrix, DamsData){
+WSM <- function(RawCriteriaMatrix, DamsDataMatrix, Ind_DamsDataMatrix){
 
 	# matrix setup
-	matrix_cols <- length(criteria_inputs) # 14 default (output size, adds summedscore)
+	matrix_cols <- length(criteria_inputs) # 14 default (output size)
 	matrix_rows <- length(available_dams) # 8 default
-	matrix_levs <- length(available_alternatives)
+	matrix_levs_ind <- length(available_alternatives)# 5 default
+	matrix_levs <- length(1:995)
 
 
-	message("Preference cols", matrix_cols, " rows ", matrix_rows, "scenarios", matrix_levs )
+	message("Decision Criteria", matrix_cols, "Dams", matrix_rows, "Decision Alternatives", matrix_levs )
 
 	#----------------------------------------
-	# Step A (PREFS): Build Preference Matrix with blank levels, no normalization
+	# Step A1 (PREFS, SINGLE DAM PROCEDURE): Build Preference Matrix with blank levels, no normalization
 	# score will be raw values from 0-1 based on user input
 	#----------------------------------------
-	PrefMatrix <- array(data = NA, c(8,14,5)) #if this needs to be 3D add ",nlevels = matrix_levs" after ncol 
-
+	Ind_PrefMatrix <- array(data = NA, c(8,14,5)) #This is a 3-D blank matrix 

```

Comment:
A better variable name or a comment could help understand use of Ind_* variables

Reply:
I will work on making this clearer. Ind_* refers to individual dams, so it references places where the third dim of the 3D matrix is 5 levels (alternatives) and not the 995 levels (scenarios) of the full multi-dam set